src/main/scala/akka/persistence/jdbc/query/JournalSequenceActor.scala

```diff
@@ -117,14 +132,12 @@ class JournalSequenceActor(readJournalDao: ReadJournalDao, config: JournalSequen
 
             // if it's a gap and has been detected before on a previous iteration we give up
             // that means that we consider it a genuine gap that will never be filled
-            case e if givenUp(e)               => missing
+            case e if (previousElement + 1 until e).forall(givenUp.contains) => missing
 
             // any other case is a gap that we expect to be filled soon
             case _ =>
               val currentlyMissing = previousElement + 1 until currentElement
-              // we don't want to declare it as missing if it has been already declared on a previous iterations
-              def alreadyMissing(e: Long) = missingByCounter.values.exists(_.contains(e))
-              missing ++ currentlyMissing.filterNot(alreadyMissing)
+              missing ++ currentlyMissing
```

Comment:
Would it make more sense if we let the `++` method decide if a missing range is already recorded or not? As such, we could remove the case on line 135. 

Basically, the code has two possible branches: 
1. there is no gap and it continues
2. there is a gap and we record it. 

As such, we can even replace the pattern matching by a simple `if/else`. 

```scala
val newMissing =
  if(previousElement + 1 == currentElement) 
    missing
  }
  else {
    val currentlyMissing = previousElement + 1 until currentElement
    missing ++ currentlyMissing
  }
```

or this:

```scala
val newMissing =
  if(previousElement + 1 == currentElement) missing
  else missing.addRange(previousElement, currentElement)
```

`addRange` would be reponsible to check if it makes sense to add the range or not.